packages/cipherstash-proxy/src/postgresql/frontend.rs

```diff
@@ -244,72 +244,97 @@ where
     async fn query_handler(&mut self, bytes: &BytesMut) -> Result<Option<BytesMut>, Error> {
         let mut query = Query::try_from(bytes)?;
 
-        let statement = self.parse_statement(&query.statement)?;
-        self.check_for_schema_change(&statement);
+        // Simple Query may contain many statements
+        let parsed_statements = self.parse_statements(&query.statement)?;
+        let mut transformed_statements = vec![];
 
-        if !eql_mapper::requires_type_check(&statement) {
-            counter!(STATEMENTS_PASSTHROUGH_TOTAL).increment(1);
-            return Ok(None);
-        }
+        warn!(target: MAPPER,
+            client_id = self.context.client_id,
+            msg = "Query contains multiple statements",
+            statements = parsed_statements.len(),
```

Comment:
Is there a check for this length being `> 1`, or is it implied some other way?

Reply:
If parsed_statements is empty, we skip the loop and essentially return immediately as a Passthrough